ambry-account/src/main/java/com/github/ambry/account/RouterStore.java

```diff
@@ -237,6 +237,10 @@ public ZNRecord update(ZNRecord znRecord) {
           // If this is not for backfill, then just read account metadata from blob, otherwise, initialize it with
           // an empty map and fill it up with the accountMap passed to constructor.
           accountMap = (!forBackFill) ? readAccountMetadataFromBlobID(blobIDAndVersion.blobID) : new HashMap<>();
+          if (accountMap == null) {
+            logAndThrowIllegalStateException("Fail to read account metadata from blob id " + blobIDAndVersion.blobID,
```

Comment:
why not do this in `readAccountMetadataFromBlobID()` ?

Reply:
I suppose we could, the readAccountMetadataFromBlobID function is also called by the fetchAccountMetadataFromZNRecord function and that function should not throw any exception. So in the readAccountMetadataFromBlobID, I catch the exception and return null.